src/Generator/ParameterGenerator.php

```diff
@@ -167,6 +167,14 @@ public function getType()
             : null;
     }
 
+    /**
+     * @return ?TypeGenerator
+     */
+    public function getTypeObject()
+    {
+        return $this->type;
+    }
```

Comment:
Exposing the type to the outside world seems like a regression here, as it will reduce the ability to change the codebase later on.